nixos/modules/services/networking/moodle-dl.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.moodle-dl;
+  moodle-dl-json = pkgs.writeText "moodle-dl.json" (builtins.toJSON cfg.config);
+in {
+  options = {
+    services.moodle-dl = {
+      enable = mkEnableOption "moodle-dl, a Moodle downloader";
+
+      config = mkOption {
+        default = {};
+        type = types.attrs;
+        description = ''
+          Configuration for moodle-dl. For a full example, see
+          <link xlink:href="https://github.com/C0D3D3V/Moodle-Downloader-2/wiki/Config.json"/>.
+        '';
+      };
+
+      startAt = mkOption {
+        type = with types; either str (listOf str);
+        description = "When to run moodle-dl. See systemd.time(7) for the format.";
+      };
+
+      directory = mkOption {
+        default = "/var/lib/moodle";
+        type = types.path;
+        description = "The path moodle-dl should download course files to.";
+      };
+
+      package = mkOption {
+        default = pkgs.moodle-dl;
+        type = types.package;
+        description = "The moodle-dl package to use.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.extraUsers.moodle = {
+      isNormalUser = false;
+      home = cfg.directory;
+      createHome = true;
+    };
+
+    system.activationScripts.moodle-dl-config = "ln -sfn ${toString moodle-dl-json} ${cfg.directory}/config.json";
+
+    systemd.services.moodle-dl = {
+      enable = true;
+      serviceConfig = {
+        Type = "oneshot";
+        User = config.users.extraUsers.moodle.name;
```

Comment:
What about `Group`?

[nobody/nogroup shouldn't be used](https://github.com/NixOS/nixpkgs/issues/55370)

Reply:
Is there something akin to `DynamicUser` for groups so that I do not have to create a group for the service manually?

Reply:
Not that I'm aware of, sorry :disappointed: